pkg/xds/envoy/secrets/v2/ca_secret.go

```diff
@@ -1,28 +1,23 @@
-package ca
+package v2
 
 import (
 	"bytes"
 
 	envoy_auth "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
 	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
 
-	sds_provider "github.com/kumahq/kuma/pkg/sds/provider"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	"github.com/kumahq/kuma/pkg/xds/envoy/tls"
 )
 
-type MeshCaSecret struct {
-	PemCerts [][]byte
-}
-
-var _ sds_provider.Secret = &MeshCaSecret{}
-
-func (s *MeshCaSecret) ToResource(name string) *envoy_auth.Secret {
+func CreateCaSecret(secret *core_xds.CaSecret) *envoy_auth.Secret {
```

Comment:
Why this is under `xds` directory? It's used once and from `sds` package

Reply:
Talked offline. sds will be moved to xds eventually. pkg/xds/envoy is a place when we try to abstract away envoy configs.